lstchain/reco/r0_to_dl1.py

```diff
@@ -449,8 +450,12 @@ def r0_to_dl1(
 
                 if not is_simu:
                     dl1_container.ucts_time = 0
-                    # Assumes R1 event source filled the "dragon" timestamp:
-                    dl1_container.dragon_time = event.trigger.time
+                    # Assumes R1 event source filled the "dragon" timestamp
+                    # as MJD; we convert back to unix timestamp:
+                    dl1_container.dragon_time = Time(event.trigger.time,
+                                                     format='mjd',
+                                                     scale='tai').to_value('unix')
```

Comment:
`trigger_time` keeps being in `mjd` format

Reply:
> `trigger_time` keeps being in `mjd` format

No, that is the "input" of the Time() call. did you test it? For me it provides the same unix timestamps as the previous lstchain code.

Reply:
I mean in the output dl1 file. Not here, sorry. It was just a side note that in the dl1 output the format is different for the two of them.

Reply:
The input of the time call is already a Time object. It is not an mjd float. It is just convereted to an mjd float by the `HDF5TableWriter`

Reply:
```
In [7]: dl1_df.dragon_time
Out[7]: 
0      1.582060e+09
1      1.582060e+09
2      1.582060e+09

In [8]: dl1_df.trigger_time
Out[8]: 
0      58897.877193
1      58897.877193
2      58897.877193

```

Reply:
@morcuended should be fixed now

Reply:
> ```
> In [7]: dl1_df.dragon_time
> Out[7]: 
> 0      1.582060e+09
> 1      1.582060e+09
> 2      1.582060e+09
> 
> In [8]: dl1_df.trigger_time
> Out[8]: 
> 0      58897.877193
> 1      58897.877193
> 2      58897.877193
> ```

Yes, I only modified dragon_time, not trigger_time. But can do both, is it used downstream?

Reply:
@moralejo I already added the `.unix` also `for trigger_time`

Reply:
now it's fine. Shall test that the first event timestamps in the dl1 file is actually the timestamps in the night summary file?

Reply:
Good test, but beware that is nanoseconds in TAI, right? And lstchain now stores UTC.

So probably the best is something like `np.isclose((Time(event_value, format='unix') - Time(summary_value / 1e9, format='unix_tai')).to_value(u.s), 0)`

Reply:
But you probably have to use the fix in `ctapipe_io_lst`, I will just add it.

Reply:
> But you probably have to use the fix in `ctapipe_io_lst`, I will just add it.

What fix? I don't see any recent one in ctapipe_io_lst

Reply:
I checked on one run (2291.0024) that the dragon_time in DL1 files is the same for this new code and for lstchain's previous one.

Reply:
> I checked on one run (2291.0024) that the dragon_time in DL1 files is the same for this new code and for lstchain's previous one.

However trigger_time is different by 6 s....  Because in old files trigger_time is 6 s behind dragon_time. I do not think we were using that trigger_time at all, while dragon_time was used, and checked. So I think that having the same values in both fields after this PR is ok.

Reply:
we were not using `trigger_time` that I remember. If I remember correctly it was not calculated in the same place that `dragon_time`, `tib_time` and `ucts_time` but in the old `_io_lst` reader (?). Not sure right now where those 6 secs come from.

Reply:
> What fix? I don't see any recent one in ctapipe_io_lst

This fix for astropy to use the correct epoch for `unix_tai`:

https://github.com/cta-observatory/ctapipe_io_lst/blob/82228781c91a697c92ace3cba024fea55123ec99/ctapipe_io_lst/event_time.py#L13-L16

Reply:
@moralejo that's the same story again with using the run start timestamp as reference for the times:

https://github.com/cta-observatory/ctapipe_io_lst/blob/dc087286fd061d1d2093022b1d6a2e6abd2b9e60/ctapipe_io_lst/__init__.py#L418-L421

Reply:
> > However trigger_time is different by 6 s.... Because in old files trigger_time is 6 s behind dragon_time (with this new code they are identical).
> 
> @moralejo that's the same story again with using the run start timestamp as reference for the times:
> 
> https://github.com/cta-observatory/ctapipe_io_lst/blob/dc087286fd061d1d2093022b1d6a2e6abd2b9e60/ctapipe_io_lst/__init__.py#L418-L421

Ok, thanks! So we stick to what we now have in this PR. Bot times the same.

Reply:
Do all of you manage to locally run pytest successfully? I have everything up to date from this PR, and consistently get a couple of errors. Below is the first one.
I interrupted it once after the first error, and it seems the directory testfiles/scripts/real_data is not created at all.

```
 def test_dl1ab_realdata():
        output_real_data_dl1ab = output_dir_realdata / “dl1ab_realdata.h5"
run_program(“lstchain_dl1ab”, “-f”, real_data_dl1_file, “-o”, output_real_data_dl1ab)
lstchain/scripts/tests/test_lstchain_scripts.py:274:
args = (‘lstchain_dl1ab’, ‘-f’, PosixPath(‘testfiles/scripts/real_data/dl1_LST-1.1.Run02008.0000_first50.h5’), ‘-o’, PosixPath(‘testfiles/scripts/real_data/dl1ab_realdata.h5’))
result = CompletedProcess(args=(‘lstchain_dl1ab’, ‘-f’, PosixPath(’testfiles/scripts/real_data/dl1_LST-1.1.Run02008.0000_first5...undError: [Errno 2] No such file or directory: \‘testfiles/scripts/real_data/dl1_LST-1.1.Run02008.0000_first50.h5\‘\n’)
```

Reply:
This is a combination of two things, @morcuended forgot to add the `@pytest.mark.private_data` to this test to mark it as one that needs the private test data.

And by default, the tests with private data are not run, so this tests fails because it depends on another one.

And we shouldn't just make tests depend on each other. We should use pytest fixtures for things we need to produce that are needed by other tests.

To really run all tests, run `pytest -m 'private_data or not private_data'`, this is what is run on the CI